openbb_terminal/common/prediction_techniques/expo_model.py

```diff
@@ -0,0 +1,131 @@
+    trend: str = "A",
+    seasonal: str = "A",
+    seasonal_periods: int = None,
+    damped: str = "F",
+    n_predict: int = 30,
+    start_window: float = 0.65,
+    forecast_horizon: int = 3,
+) -> Tuple[List[float], List[float], Any, Any]:
+
+    """Performs Probabalistic Exponential Smoothing forecasting
+    This is a wrapper around statsmodels Holt-Winters' Exponential Smoothing;
+    we refer to this link for the original and more complete documentation of the parameters.
+
+    https://unit8co.github.io/darts/generated_api/darts.models.forecasting.exponential_smoothing.html?highlight=exponential
+
+    Parameters
+    ----------
+    data : Union[pd.Series, np.ndarray]
+        Input data.
+    trend: str
+        Trend component.  One of [N, A, M]
+        Defaults to ADDITIVE.
+    seasonal: str
+        Seasonal component.  One of [N, A, M]
+        Defaults to ADDITIVE.
+    seasonal_periods: int
+        Number of seasonal periods in a year
+        If not set, inferred from frequency of the series.
+    damped: str
+        Dampen the function
+    n_predict: int
+        Number of days to forecast
+    start_window: float 
+        Size of sliding window from start of timeseries and onwards
+    forecast_horizon: int
+        Number of days to forecast when backtesting and retraining historical
+
+    Returns
+    -------
+    List[float]
+        Adjusted Data series
+    List[float]
+        List of predicted values
+    Any
+        Fit Prob. Expo model object.
+    """
+
+    filler = MissingValuesFiller()
+    data["date"] = data.index  # add temp column since we need to use index col for date
+    ticker_series = TimeSeries.from_dataframe(
+        data,
+        time_col="date",
+        value_cols=["AdjClose"],
+        freq="B",
+        fill_missing_dates=True,
+    )
+
+    ticker_series = filler.transform(ticker_series)
+    ticker_series = ticker_series.astype(np.float32)
+    train, val = ticker_series.split_before(0.85)
+
+    if trend == "M":
+        trend = ModelMode.MULTIPLICATIVE
+    elif trend == "N":
+        trend = ModelMode.NONE
+    else:  # Default
+        trend = ModelMode.ADDITIVE
+
+    if seasonal == "M":
+        seasonal = SeasonalityMode.MULTIPLICATIVE
+    elif seasonal == "N":
+        seasonal = SeasonalityMode.NONE
+    else:  # Default
+        seasonal = SeasonalityMode.ADDITIVE
+
+    damped = True if damped == "T" else False
+
+    # Model Init
+    model_es = ExponentialSmoothing(
+        trend=trend, seasonal=seasonal, seasonal_periods=seasonal_periods, damped=damped
+    )
+
+    # Training model based on historical backtesting
+    historical_fcast_es = model_es.historical_forecasts(
+        ticker_series,
+        start=start_window,
+        forecast_horizon=forecast_horizon,
+        verbose=True,
+    )
+
+    # Show forecast over validation # and then +n_predict afterwards sampled 10 times per point
+    probabilistic_forecast = model_es.predict(n_predict, num_samples=10)
```

Comment:
10 samples is somewhat on the low side - something like 500 would seem more reasonable to estimate the distribution. Generating these samples is fast in Numpy so it shouldn't incur any noticeable penalty, and it'll improve precision (especially as later you need the 10th and 90th percentiles).

Reply:
Great, will update. Thanks for the insight.

Reply:
Thanks for the feedback @hrzn ❤️ 

We are going to revamp our entire prediction menu with Darts and create a release around it 🚀 🚀

Reply:
Awesome :)